src/core/models/filter.js

```diff
@@ -23,6 +23,34 @@ export default class Filter {
     }
   }
 
+  /**
+   * Check whether this simple filter is equal to another simple filter.
+   * @param {Filter} filter
+   * @returns {boolean}
+   */
+  isEqualToSimpleFilter (filter) {
+    if (this.getFilterKey() !== filter.getFilterKey()) {
```

Comment:
Since we're only checking one-level deep, I think we can simplify this method a bit. I think instead of your for loop, we could do something like fieldIds1.any(id -> !fieldIdsToValues2.hasOwnProperty(id) || fieldIdsToValues2[id] !== fieldIdsToValues1[id]).

Reply:
Yes you're right. I was worried about duplicate entries, so if you had something like
```js
{
  a: 1,
  a: 1,
  b: 2
},
{
  a: 1,
  b: 2,
  b: 2,
}
```

these are definitely equal, though, even if the objects themselves are not equal